Fixing the Windows nested package issues with running Nightwatch and Jest

BenBoersma opened the pull request:
### Summary
Resolves https://github.com/cerner/terra-core/issues/815, this fixes the pathing issues trying to run nightwatch, but nightwatch itself still does not run.
Resolves https://github.com/cerner/terra-core/issues/813

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

mjhenkes commented on packages/terra-props-table/package.json:
```diff
@@ -44,7 +44,6 @@
-    "test:wdio": "../../node_modules/.bin/wdio ../../node_modules/terra-dev-site/config/wdio/wdio.conf.js"
-  }
+    "test:jest": "node ../../scripts/run-jest/index.js",
+    "test:wdio": "../../node_modules/.bin/wdio ../../node_modules/terra-dev-site/config/wdio/wdio.conf.js"  }
```
move curly brace to next line

mjhenkes commented on scripts/run-jest/index.js:
```diff
@@ -0,0 +1,20 @@
+const spawn = require('child_process').spawn;
```
What is this file (add comments)? And if it's common it should live somewhere common so we don't have to duplicate it across repos.

BenBoersma replied:
Its a nodejs module, https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options

BenBoersma commented:
@mjhenkes 
I just checked and indeed it is having that issue.
Specifically went to "packages/terra-card" and ran "test:wdio: and got 

> '..' is not recognized as an internal or external command,

emilyrohrbough commented on scripts/run-jest/index.js:
```diff
@@ -0,0 +1,20 @@
+const spawn = require('child_process').spawn;
+const os = require('os');
```
I notice we didn't add this as a dependency in our root level package.json

emilyrohrbough commented on scripts/run-jest/index.js:
```diff
@@ -0,0 +1,20 @@
+    regex = `${process.env.npm_package_name}\\\\tests`;
+    jestcmd = 'jest.cmd';
+    break;
+  default:
```
I am worried this check is too specific? What are the possible type returns? could we check for windows and if not windows use `jestcmd= 'jest' by default?

emilyrohrbough replied:
Is the `.cmd` a by product of all bin scripts on windows machines? If so we might look into opening this up to accept a cmd paramater to toggle `jest || jest.cmd` or for example `wdio || wdio.cmd`

emilyrohrbough replied:
Does this command correctly pass command line arguments ? For example, running from terra-button, does this run?

```bash
npm run tt:wdio -- -u
```

BenBoersma commented:
Changed the way I fixed the issue with Jest. Now instead of creating a custom script to run jest, I have switch jestconfig.json to jestconfig.js and updated its testMatch regex to dynamically match the directory it is called from.

mjhenkes commented on jestconfig.js:
```diff
@@ -0,0 +1,16 @@
+module.exports = {
```
Fun Fact: We have two jest configs floating around in our repo. This one is used when running jest in the individual packages, one in package.json is run when running jest on the lerna repo overall?
Two questions,
1. Does running jest at the main repo have an issue on PC?
2. Could you consolidate these configs (I think you could just delete the config from package.json)

![rockfact](https://user-images.githubusercontent.com/1588747/40856826-ce411a9c-659e-11e8-8591-ce5bd1b4f947.png)

BenBoersma replied:
1. Running Jest on the main repo has no issue on PC.

2. Looking into it now
